cluster/addons/dns/README.md

```diff
@@ -59,6 +59,21 @@ When enabled, pods are assigned a DNS A record in the form of `pod-ip-address.my
 
 For example, a pod with ip `1.2.3.4` in the namespace `default` with a dns name of `cluster.local` would have an entry: `1-2-3-4.default.pod.cluster.local`.
 
+
+####A Records and hostname Based on Pod Annotations - A Beta Feature in Kubernetes v1.3
+Currently when a pod is created it's hostname is the Pod's Name value.
+With v1.3, users can specify a Pod Annotation, pod.beta.kubernetes.io/hostname, to specify what the Pod's hostname should be.
+If the annotation is specified, the annotation value takes precendence over the Pod's Name, to be the hostname of the pod.
+
+v1.3 introduces a beta feature where the user can specify a Pod Annotation, pod.beta.kubernetes.io/subdomain, to specify what the Pod's subdomain should be.
+If the annotation is specified, the fully qualified Pod hostname will be <hostname>.<subdomain>.<pod namespace>.svc.<cluster domain>
+
+If there exists a headless service in the same namespace as the pod and with the same name as the subdomain, the cluster's KubeDNS Server will also return an A record for the Pod's fully qualified hostname.
+
+With v1.3, the Endpoints object also has a new Annotation endpoints.beta.kubernetes.io/hostnames-map. Its value is the json representation of map[string(IP)][endpoints.HostRecord]
+For example: '{"10.245.1.6":{HostName: "my-webserver"}}'. If the Endpoints are for a headless service, then A records will be created with the format <hostname>.<subdomain>.<pod namespace>.svc.<cluster domain>
+For the example json, the subdomain would be headless service name, and the hostname would be  "my-webserver" and the A record lookup would return 10.245.1.6
```

Comment:
I would add that this is not generally needed for end-users but is used by the internal service controller to deliver the aforementioned feature.